src/LuaMultiThreaded2.cpp

```diff
@@ -88,19 +88,24 @@ extern "C" static int mutexNew(LuaState * aState)
 
 
 
-/** */
-extern "C" static int mutexLock(LuaState * aState)
+/** Executes the provided function while the mutex is locked. */
+extern "C" static int mutexDoWhileLocked(LuaState * aState)
 {
 	auto mutexObj = reinterpret_cast<std::mutex**>(luaL_checkudata(aState, 1, MUTEX_METATABLE_NAME));
 	if (mutexObj == nullptr)
 	{
 		luaL_argerror(aState, 0, "'mutex' expected");
 		return 0;
 	}
+	if (*mutexObj == nullptr) 
+	{
+		luaL_argerror(aState, 0, "'mutex' expected");
```

Comment:
I wasn't really sure what kind of error message I should put here. Any suggestions?

Reply:
This is probably unreachable by a user-code error, but rather is some kind of internal error only. We basically don't ever expect this to happen, but we want to make sure we error out instead of hard-crashing, if it does happen. So I think something like this will do:
```suggestion
		luaL_error(aState, "Internal error: 'mutex' object is invalid");
```